units/src/amount/mod.rs

```diff
@@ -38,6 +38,48 @@ pub use self::{
     unsigned::Amount,
 };
 
+/// Result of an operation on [`Amount`] or [`SignedAmount`].
+///
+/// The type parameter `T` should be normally `Amout` or `SignedAmount`.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum AmountOpResult<T> {
```

Comment:
Should we seal this so that `T` is guaranteed to only be `Amount` or `SignedAmount`? 

Re the name:

- re `Amount` - only if this is never going to be used for other types (weight, fee rate) - discussion open still.
- re `Op` - only if this is never going to be used to implement a trait outside of `core::ops` - correct?

Reply:
I don't think we should "seal" it, I'm thinking of having this too: `fn checked_add_assign(&mut self, rhs: Amount) -> NumOpResult<()>`. Then one day when `Try` is stable we'll be able to use `accum.checked_add_assign(partial_sum)?`.

I'd slightly prefer `Num` even if only `Amount` produces these because it's shorter and we might add a new thing at some point that will need it.

`Op` refers to "mathematical operation", so returning it from various helper methods (`sum`) is fine. It's more like "don't use it for conversions&shit" But I suspect `From<OutOfRange> for AmountOpError` should exist.

Reply:
Flagging that #4056 does not implement `checked_add_assign` and I did not raise an issue to do so. Please raise one if you still want this.

Reply:
I guess since `Try` is not stable, it's not worth it yet.